File: leetcode/hashMap/1475.cpp
#include <algorithm>
#include <functional> //包含std::greater 和 std::less
#include <iostream>
#include <limits.h>
#include <map>
#include <math.h>
#include <queue>
#include <stack>
#include <unordered_set>
#include <vector>

using namespace std;
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};
struct ListNode {
  int val;
  ListNode *next;
  ListNode() : val(0), next(nullptr) {}
  ListNode(int x) : val(x), next(nullptr) {}
  ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
  vector<int> finalPrices(vector<int> &prices) {
    vector<int> result(prices);
    int n = result.size();
    for (int i = 0; i < n - 1; i++) {
      for (int j = i + 1; j < n; j++) {
        if (result[j] <= result[i]) {
          result[i] -= result[j];
          break;
        }
      }
    }
    return result;
  }
};
int main() {
  // 示例二叉树
  Solution solution;
  vector<int> nums = {7, 1, 5, 3, 6, 4};
  vector<int> result = solution.finalPrices(nums);
  // std::cout << "result: " << result << std::endl;
}

File: leetcode/hashMap/1475.py
from typing import List
from queue import Queue


class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        result = prices[:]
        n: int = len(result)
        for i in range(n - 1):
            for j in range(i + 1, n):
                if result[j] <= result[i]:
                    result[i] -= result[j]
                    break
        return result


if __name__ == "__main__":
    solution = Solution()
    nums = [8, 4, 6, 2, 3]
    result = solution.finalPrices(nums)
    print(f"result = {result}")
